server/services/unlockListService.ts

```diff
@@ -100,13 +101,23 @@ export default class UnlockListService {
       const appointments = scheduledEvents?.appointments
         .filter(app => app.prisonerNumber === prisoner.prisonerNumber)
         .filter(app => applyCancellationDisplayRule(app))
-      const courtHearings = scheduledEvents?.courtHearings.filter(crt => crt.prisonerNumber === prisoner.prisonerNumber)
-      const visits = scheduledEvents?.visits.filter(vis => vis.prisonerNumber === prisoner.prisonerNumber)
-      const adjudications = scheduledEvents?.adjudications.filter(adj => adj.prisonerNumber === prisoner.prisonerNumber)
-      const transfers = scheduledEvents?.externalTransfers.filter(tra => tra.prisonerNumber === prisoner.prisonerNumber)
+        .filter(app => (cancelledEventsFilter === YesNo.YES ? app : !app.cancelled))
```

Comment:
is this easier to read? 

.filter(app => !app.cancelled || cancelledEventsFilter === YesNo.YES)